MIGRATION.md

```diff
@@ -743,36 +747,102 @@ Instead, use `export const parameters = {};` and `export const decorators = [];`
+
+- `docs.inlineStories` has been renamed `docs.story.inline`
+- `docs.iframeHeight` has been renamed `docs.story.iframeHeight`
+- `notes` and `info` are no longer supported, instead use `docs.description.story | component`
+
+#### MDX docs files
+
+Previously `.stories.mdx` files were used to both define and document stories. In 7.0, we have deprecated defining stories in MDX files, and consequently have changed the suffix to simply `.mdx`. Our default `stories` glob in `main.js` will now match such files -- if you want to write MDX files that do not appear in Storybook, you may need to adjust the glob accordingly.
+
+If you were using `.stories.mdx` files to write stories, we encourage you to move the stories to a CSF file, and *attach* an `.mdx` file to that CSF file to document them. You can use the `Meta` block to attach a MDX file to a CSF file, and the `Story` block to render the stories:
 
 ```mdx
-import { Meta } from '@storybook/blocks';
+import { Meta, Story } from '@storybook/blocks';
 import * as ComponentStories from './some-component.stories';
 
 <Meta of={ComponentStories} />
+
+<Story of={ComponentStories.Primary} />
 ```
 
-You can create as many docs entries as you like for a given component. Note that if you attach a docs entry to a component it will replace the automatically generated entry from `DocsPage` (See below).
+You can create as many docs entries as you like for a given component. Note that if you attach a docs entry to a component it will replace the automatically generated entry from Autodocs.
 
 By default docs entries are listed first for the component. You can sort them using story sorting.
 
-#### Referencing stories in docs files
+
+#### Unattached docs files
+
+In Storybook 6.x, to create a unattached docs MDX file (that is, one not attached to story or a CSF file), you'd have to create a `.stories.mdx` file, and describe its location with the `Meta` doc block:
+
+```mdx
+import { Meta } from '@storybook/addon-docs';
+
+<Meta title="Introduction" />
+```
+
+In 7.0, things are a little simpler -- you should call the file `.mdx` (drop the `.stories`). This will mean behind the scenes there is no story attached to this entry. You may also drop the `title` and use autotitle (and leave the `Meta` component out entirely, potentially).
+
+
+#### Doc Blocks
+
+Additionally to changing the docs information architecture, we've updated the API of the doc blocks themselves to be more consistent and future proof.
+
+**General changes**
+
+- Each block now uses `of={}` as a primary API -- where the argument to the `of` prop is a CSF or story *export*.
+
+- When you've attached to a CSF file (with the `Meta` block, or in Autodocs), you can drop the `of` and the block will reference the first story or the CSF file as a whole.
```

Comment:
When reading this I would expect `Story` without an `of` to select the primary story in an attached file, but currently `Story` throws in that scenario. What's correct here?

Reply:
I'm happy with either, simple enough to change. WDYT @shilman?

Reply:
Will do in a separate PR!

Reply:
https://github.com/storybookjs/storybook/pull/20765